be/src/exec/pipeline/exchange/local_exchange.cpp

```diff
@@ -338,6 +338,26 @@ Status PassthroughExchanger::accept(const ChunkPtr& chunk, const int32_t sink_dr
     return Status::OK();
 }
 
+Status ConnectorSinkPassthroughExchanger::accept(const ChunkPtr& chunk, const int32_t sink_driver_sequence) {
+    size_t sources_num = _source->get_sources().size();
+    if (sources_num == 1) {
+        _source->get_sources()[0]->add_chunk(chunk);
+    } else {
+        // Scale up writers when current buffer memory utilization is more than 50% of the maximum and data processed
```

Comment:
what's the thought process behind the number `50%`?

Reply:
To be honest, this value is just the same as Trino, and our tests indicate that 50% is suitable.

Reply:
do you think it's better that reset _data_processed when new writer join in, then next time we check from new writer come in if each write have get more than threshold data size?

Reply:
Sure, however there seems little difference right? Our purpose is to scale up as quick as possible when it comes to large data, so maybe it's not necessary to be that accurate.